### PR TITLE
Improve the doc for federated identities

### PR DESCRIPTION
**Pull request by powerful23**
*Issue #, if available:*

Customers are confused about how to do federation with the Cognito Service because it has two different concepts: federated with User Pool and federated with Identity Pool. This change aims to make it clearer in the doc and help the customer to get the correct place to setup once they know which service they are using.

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: js/authentication.md [hunk 1]
```diff
@@ -549,7 +549,7 @@ this._validAuthStates = ['signedIn'];
 
 Then, in the component's constructor,  implement `showComponent(theme) {}` in lieu of the typical `render() {}` method.
 
-### Federated Identities (Social Sign-in)
+### Federated with Cognito Identities
```
Comment (undefobj) [inline js/authentication.md]: Can you update to "Federate with Cognito Identity Pools"

File: js/authentication.md [hunk 2]
```diff
@@ -1075,9 +1075,9 @@ const federated = {
 
 There is also `withGoogle`, `withFacebook`, `withAmazon` components, in case you need to customize a single provider.
 
-### Using Amazon Cognito Hosted UI
+### Federated with Cognito User Pool Service
```
Comment (undefobj) [inline js/authentication.md]: Can you update to "Federate with Cognito User Pools"

File: js/authentication.md (hunk 1 above)
Comment (powerful23) [reply js/authentication.md]: Done.

File: js/authentication.md (hunk 2 above)
Comment (powerful23) [reply js/authentication.md]: Done

File: js/authentication.md [hunk 3]
```diff
@@ -1075,9 +1075,9 @@ const federated = {
...
 
-Amazon Cognito provides a customizable user experience via the hosted UI. The hosted UI supports OAuth 2.0 and Federated Identities with Facebook, Amazon, Google, OIDC and SAML providers. To learn more about Amazon Cognito Hosted UI, please visit [Amazon Cognito Developer Guide](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-configuring-app-integration.html).
+Amazon Cognito User Pool Service provides a customizable user experience via the hosted UI which supports OAuth 2.0 and Federated Identities with Facebook, Amazon, Google, OIDC and SAML providers. To learn more about Amazon Cognito Hosted UI, please visit [Amazon Cognito Developer Guide](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-configuring-app-integration.html).
```
[the remark this answers is not in the document]
Comment (powerful23) [reply js/authentication.md]: Great. Will change it accordingly.

**Pull request merged**